ambry-clustermap/src/test/java/com.github.ambry.clustermap/HelixClusterManagerTest.java

```diff
@@ -325,6 +331,48 @@ public void clientInitiatedLivenessChangeTest() {
     assertStateEquivalency();
   }
 
+  /**
+   * Test that {@link ResponseHandler} works as expected in the presence of various types of server events. The test also
+   * verifies the states of datanode, disk and replica are changed correctly based on server event.
+   */
+  @Test
+  public void onServerEventTest() {
+    if (useComposite) {
+      return;
+    }
+    // Test configuration: we select the disk from one datanode and select the replica on that disk
+
+    // Initial state: disk is down; Server event: Replica_Unavailable; Expected result: disk becomes available again
+    mockServerEventsAndVerify(
+        new ResourceState[]{ResourceState.Node_Up, ResourceState.Disk_Down, ResourceState.Replica_Down},
```

Comment:
these initial states need to change. Resource state of Replica is up here - resource state of disk is down